pkgs/by-name/ei/eiffelstudio/package.nix

```diff
@@ -0,0 +1,125 @@
+{
+  copyDesktopItems,
+  fetchurl,
+  lib,
+  makeDesktopItem,
+  makeWrapper,
+  pkgs,
+  stdenv,
+
+  dotnetSupport ? false, # Untested integration with .NET, disabled for now
+}:
+
+let
+  platformName =
+    {
+      "x86_64-linux" = "linux-x86-64";
+      "aarch64-linux" = "linux-arm64";
+      "x86_64-darwin" = "macosx-x86-64"; # Doesn't compile with clang yet
+      "aarch64-darwin" = "macosx-armv6"; # Need to do more testing on macOS
+    }
+    .${stdenv.system} or null;
+in
+stdenv.mkDerivation rec {
+  pname = "eiffelstudio";
+  version = "25.02.98732";
+
+  src = fetchurl {
+    url = "https://ftp.eiffel.com/pub/download/latest/pp/PorterPackage_${lib.versions.majorMinor version}_rev_${lib.versions.patch version}.tar";
+    sha256 = "sha256-/XoewqCeh1NfB3ve9UL+0WZfZ5DEa4N7REl67Ftlxt0=";
+  };
+
+  nativeBuildInputs =
+    with pkgs;
+    [
+      coreutils
+      copyDesktopItems
+      gnused
+      makeWrapper
+      pax
+      pkg-config
+    ]
+    ++ lib.optional dotnetSupport dotnet-sdk;
+
+  buildInputs =
+    with pkgs;
+    [
+      adwaita-icon-theme
+      cairo
+      gdk-pixbuf
+      glib
+      gtk3
+      pango
+      pkg-config
+      xorg.libX11
+      xorg.libXtst
+    ]
+    ++ lib.optional dotnetSupport dotnet-runtime;
+
+  sourceRoot = "PorterPackage";
+  #patches = [ ./build-debugging.patch ]; # Leaving this here in case I need it later
+  postPatch = ''
+    # A script inside this tarball depends on hardcoded coreutils + gnused paths
+    tar -xf c.tar.bz2
+
+    # All of these (not shebang) references to /bin must be replaced
+    for i in cp ln mv rm; do
+      substituteInPlace C/CONFIGS/${platformName} --replace-fail "/bin/$i" "${pkgs.coreutils}/bin/$i"
+    done
+    substituteInPlace C/CONFIGS/${platformName} --replace-fail "/bin/sed" "${pkgs.gnused}/bin/sed"
+
+    # Re-assemble the tarball with fixed up paths
+    rm c.tar.bz2
+    tar -cjSf c.tar.bz2 C
+  '';
+
+  hardeningDisable = [ "format" ]; # Allow printf without a format string
+  # Better explained here: https://fedoraproject.org/wiki/Format-Security-FAQ
+  buildPhase = ''
+    runHook preBuild
+    ./compile_exes ${platformName}
```

Comment:
A shame that upstream does not offer a standard build system....

Reply:
Yes, it is completely homespun and kind of a pain.